sdk/core/core-http/src/serializer.ts

```diff
@@ -1231,37 +1348,92 @@ export interface BaseMapper {
   constraints?: MapperConstraints;
 }
 
+/**
+ * Mappers are definitions of the Data models used in the library.
+ * These data models are part of the Operation or Client definitions in the responses or parameters.
+ */
 export type Mapper = BaseMapper | CompositeMapper | SequenceMapper | DictionaryMapper | EnumMapper;
 
+/**
+ * Helps identify how to transform from one set of properties in the source object, to a set of properties in the target object.
+ */
 export interface PolymorphicDiscriminator {
+  /**
+   * Name used during serialization.
+   */
   serializedName: string;
+  /**
+   * Name of the client.
+   */
   clientName: string;
+  /**
+   * It may contain any other property.
+   */
   [key: string]: string;
 }
 
+/**
+ * A mapper composed of other mappers.
+ */
 export interface CompositeMapper extends BaseMapper {
+  /**
+   * The type descriptor of the `CompositeMapper`.
+   */
   type: CompositeMapperType;
 }
 
+/**
+ * A mapper describing a sequence.
+ */
 export interface SequenceMapper extends BaseMapper {
+  /**
+   * The type descriptor of the `SequenceMapper`.
+   */
   type: SequenceMapperType;
 }
 
+/**
+ * A mapper describing a dictionary.
```

Comment:
this really means key/value pairs, but they're a JS object rather than a Map<> or something